### PR TITLE
Add XNNPack build on Linux ARM64 and improve Linux CPU

### PR DESCRIPTION
**Pull request by mszhanyi**
### Description
1. Add XNNPack build on Linux ARM64
2. Build only one python wheel for PR request.
[AB#49763](https://aiinfra.visualstudio.com/6a833879-cd9b-44a4-a9de-adc2d818f13c/_workitems/edit/49763)

### Motivation and Context
Why I add xnnpack build on Linux ARM64  rather than Windows ARM64.
Becuase KleidiAI  doesn't support Windows

```
IF(XNNPACK_TARGET_PROCESSOR STREQUAL "arm64" AND XNNPACK_ENABLE_ARM_I8MM AND NOT CMAKE_C_COMPILER_ID STREQUAL "MSVC")
  IF (XNNPACK_ENABLE_KLEIDIAI)
    MESSAGE(STATUS "Enabling KleidiAI for Arm64")
  ENDIF()
ELSE()
  SET(XNNPACK_ENABLE_KLEIDIAI OFF)
ENDIF()
```

### CODE REVIEW

File: tools/ci_build/github/linux/build_linux_python_package.sh [hunk 1]
```diff
@@ -89,10 +89,13 @@ export CMAKE_ARGS="-DONNX_GEN_PB_TYPE_STUBS=ON -DONNX_WERROR=OFF"
 for PYTHON_EXE in "${PYTHON_EXES[@]}"
 do
   rm -rf /build/"$BUILD_CONFIG"
-  ${PYTHON_EXE} -m pip install -r /onnxruntime_src/tools/ci_build/github/linux/python/requirements.txt  
-  ${PYTHON_EXE} /onnxruntime_src/tools/ci_build/build.py "${BUILD_ARGS[@]}"
-
+  # that's a workaround for the issue that there's no python3 in the docker image
```
Comment (snnn) [inline tools/ci_build/github/linux/build_linux_python_package.sh:92]: Why this workaround is needed? The current code works well.

Comment (snnn) [reply tools/ci_build/github/linux/build_linux_python_package.sh:92]: Oh, I see. Because the python3 is not in PATH? But you can add the folder to PATH.

Comment (mszhanyi) [reply tools/ci_build/github/linux/build_linux_python_package.sh:92]: No python3 in the docker image. xnnpack will fail without it.
https://github.com/google/XNNPACK/blob/23efce6d7b98c0d1ac15d615d74415dd2feb72b8/CMakeLists.txt#L405-L409C2
```
MESSAGE(STATUS "Generating microkernels.cmake")
EXECUTE_PROCESS(
  COMMAND python3 "${PROJECT_SOURCE_DIR}/tools/update-microkernels.py" --output "${PROJECT_BINARY_DIR}"
  RESULT_VARIABLE UPDATE_MICROKERNELS_RESULT
)
```

Comment (mszhanyi) [reply tools/ci_build/github/linux/build_linux_python_package.sh:92]:
```
-- Generating microkernels.cmake
CMake Error at /build/Release/_deps/googlexnnpack-src/CMakeLists.txt:346 (MESSAGE):
  Failed to generate "microkernels.cmake".
```

Comment (mszhanyi) [reply tools/ci_build/github/linux/build_linux_python_package.sh:92]: There's no code to build xnnpack in packaging docker(manylinux).

Comment (snnn) [reply tools/ci_build/github/linux/build_linux_python_package.sh:92]: But `${PYTHON_EXE}`'s parent folder has python3 . You could just add that folder to PATH instead.

For example, "/opt/python/cp313-cp313/bin" has

```
idle3     pip   pip3.13  pydoc3.13        python   python3.13         python3-config
idle3.13  pip3  pydoc3   pyproject-build  python3  python3.13-config
```

Comment (mszhanyi) [reply tools/ci_build/github/linux/build_linux_python_package.sh:92]: > But `${PYTHON_EXE}`'s parent folder has python3 . You could just add that folder to PATH instead.
> 
> For example, "/opt/python/cp313-cp313/bin" has
> 
> ```
> idle3     pip   pip3.13  pydoc3.13        python   python3.13         python3-config
> idle3.13  pip3  pydoc3   pyproject-build  python3  python3.13-config
> ```

thanks. updated

**Review by snnn — APPROVED** [review]

**Pull request merged**